cpp/src/arrow/util/async_generator.h

```diff
@@ -147,6 +147,51 @@ Future<std::vector<T>> CollectAsyncGenerator(AsyncGenerator<T> generator) {
   return Loop(LoopBody{std::move(generator), std::move(vec)});
 }
 
+/// \brief this is just like a MapGenerator but the map fun returns a thing instead of a
+/// future
```

Comment:
Can you expand on this description?  I'm pretty sure the important part is that you are launching a new thread task with each input right?

Reply:
That is right.